hack/verify-govet.sh

```diff
@@ -23,8 +27,12 @@ function govet_blacklist_contains() {
 	return 1
 }
 
-for test_dir in $(os::util::list_go_src_dirs); do
-	if ! result="$(go tool vet -shadow=false -printfuncs=Info,Infof,Warning,Warningf "${test_dir}" 2>&1)"; then
+# This is required before we run govet for the results to be correct.
+# See https://github.com/golang/go/issues/16086 for details.
+go install ./cmd/...
```

Comment:
this is what's needed but it fails for me on CGO missing headers...
```
go install ./cmd/...
# github.com/openshift/origin/vendor/github.com/mtrmac/gpgme
vendor/github.com/mtrmac/gpgme/data.go:4:11: fatal error: gpgme.h: No such file or directory
 // #include <gpgme.h>
           ^~~~~~~~~
compilation terminated.
```

Reply:
no, there are build tags to exclude this

Reply:
@mfojtik could you elaborate? that's what happens if I try it locally

Reply:
You still need the headers if you're building origin, so that's not a problem.

Reply:
I never needed them on make though. Is there no way to vendor that?

Reply:
@soltysh I think this need to be 
```bash
go install -tags="${OS_GOFLAGS_TAGS-}" ./cmd/... 
```
you can find that constant here 
https://github.com/openshift/origin/blob/a089c5712208b48640d3dee7f10c1dd47bc36322/hack/lib/constants.sh#L13
 This should also go to the go vet args.


as this is only a hack till our oldest supported version is based on go >=1.10 (I want this to backport ideally) not splitting the whole build process into install func and wiring it here is fine. https://github.com/openshift/origin/blob/d0c7303ee6b6edfa9af115f1abca222b976987a3/hack/lib/build/binaries.sh#L236-L242

Reply:
@soltysh isn't this outstanding? Why not use the `os::build` functions instead of threading this together yourself here?

Reply:
lemme check...